lib/gooddata/models/user_filters/user_filter_builder.rb

```diff
@@ -542,7 +543,7 @@ def self.normalize_filters(filters)
     def self.sanitize_filters_to_delete(to_delete, users_brick_input, project_users)
       return to_delete unless users_brick_input && users_brick_input.any?
       user_profiles = users_brick_input.map do |user|
-        result = project_users.find { |u| u.login == user['login'] }
+        result = project_users.find { |u| u.login == user.with_indifferent_access['login'] }
```

Comment:
this is not preventing us from making a similar mistake in the future. How about we turn all results of the realize() call into hashes with indifferent access, so the access will be uniform for reading all sources (CSV, ADS, S3..) ?

Reply:
`#realize` does not return a hash. It returns a file path which is then read in inconsistent ways. Created a ticket for fixing the inconsistencies. https://jira.intgdc.com/browse/TMA-838